pallets/cash/src/internal/events.rs

```diff
@@ -513,6 +510,84 @@ pub fn receive_chain_reorg<T: Config>(
 mod tests {
     use super::*;
     use crate::tests::*;
+    use sp_core::offchain::testing;
+    use sp_core::offchain::OffchainExt;
+    use ethereum_client::EthereumBlock;
+    use params;
+
+    fn gen_old_blocks(start_block: u64, end_block: u64, start_hash: u8) -> Vec<EthereumBlock> {
+        let mut next_hash = start_hash;
+        let mut v: Vec<ethereum_client::EthereumBlock> = vec![];
+        for i in start_block..=end_block {
```

Comment:
Is `end_block` expected to be inclusive range in the context of this function? I see the callsite and it looks like it is somewhat double counting ? `common_ancestor_idx.clone() - 1`

Reply:
yes good catch